specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/firewallPolicy.json

```diff
@@ -1464,6 +1714,236 @@
+    "IDPSQueryObject": {
+      "description": "Will describe the query to run against the IDPS signatures DB",
+      "type": "object",
+      "properties": {
+        "filters": {
+          "type": "object",
+          "description": "Contain all filters names and values",
+          "$ref": "#/definitions/Filters"
+        },
+        "search": {
+          "type": "string",
+          "description": "Search term in all columns"
+        },
+        "orderBy": {
+          "type": "object",
+          "$ref": "#/definitions/OrderBy",
+          "description": "Column to sort response by"
+        },
+        "resultsPerPage": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 1,
+          "maximum": 1000,
+          "description": "The number of the results to return in each page"
+        },
+        "skip": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of records matching the filter to skip"
+        }
+      }
+    },
+    "Filters": {
+      "description": "Describers the filters to filter response by",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/FilterItems"
+      }
+    },
+    "FilterItems": {
+      "description": "Will contain the filter name and values to operate on",
+      "type": "object",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "The name of the field we would like to filter"
+        },
+        "values": {
+          "type": "array",
+          "description": "List of values to filter the current field by",
+          "items": {
+            "type": "string",
+            "description": "Value of the field to filter by"
+          }
+        }
+      }
+    },
+    "OrderBy": {
+      "description": "Describes a column to sort",
+      "type": "object",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "Describes the actual column name to sort by"
+        },
+        "asc": {
+          "type": "boolean",
+          "description": "Describes if results should be in ascending order"
+        }
+      }
+    },
+    "QueryResults": {
+      "description": "Query result",
+      "type": "object",
+      "properties": {
+        "allRecords": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Number of total records matching the query."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "When present, the value can be passed to a subsequent query call (together with the same query and scopes used in the current request) to retrieve the next page of data."
+        },
+        "signatures": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SingleQueryResult"
+          },
+          "description": "Array containing the results of the query"
+        }
+      }
+    },
+    "SingleQueryResult": {
```

Comment:
>SingleQueryResult [](http://example.com/codeflow?start=5&length=17)

Do mode, severity, and direction need to be integers rather than string enum so the values include their meaning?

Reply:
Yes, this allows us to order using these values.